nipype/interfaces/spm/preprocess.py

```diff
@@ -217,6 +217,15 @@ def _parse_inputs(self):
         """validate spm realign options if set to None ignore
         """
         einputs = super(Realign, self)._parse_inputs()
+        is_gz = lambda x: x.endswith('.gz')
+        if isdefined(self.inputs.in_files):
+            if isinstance(self.inputs.in_files, list):
+                for imgf in self.inputs.in_files:
+                    if is_gz(imgf):
+                        raise TypeError('Input files must be uncompressed')
+            else:
+                if is_gz(self.inputs.in_files):
+                    raise TypeError('Input files must be uncompressed')
```

Comment:
Not sure if there's a better approach in general, but the test can be made much more concise:

```Python
if isdefined(self.inputs.in_files) and any(imgf.endswith('.gz')
                                           for imgf in filename_to_list(self.inputs.in_files)):
    raise TypeError('Input files must be uncompressed')
```